### PR TITLE
bootstrap: Fix all the pep-8 issues reported by flake8

### PR DESCRIPTION
**Pull request by milmazz**
This commit also adds a few missing docstrings.

Today, after reading this [article](https://blog.rust-lang.org/2017/06/27/Increasing-Rusts-Reach.html), I downloaded this project and started building from source. In the meantime, I began to read the `bootstrap.py`, to know more about the building process, and I made a few changes, this is my first contribution to the project, hope you like it.

BTW, I have a few doubts about the `bootstrap.py`, any guidance is more than welcome:

* Where can I find the unit tests for this script? In case it doesn't exist yet, do you like to include some unit tests with pytest?
* Some methods like `fix_executable`, `get_string`, and `exe_suffix` in the `RustBuild` class should be converted to a function because it doesn't use `self` anywhere. What do you think?

### CODE REVIEW

File: src/bootstrap/bootstrap.py [hunk 1]
```diff
@@ -245,7 +250,8 @@ def fix_executable(self, fname):
             return
 
         # At this point we're pretty sure the user is running NixOS
-        print("info: you seem to be running NixOS. Attempting to patch " + fname)
+        nix_os_msg = "info: you seem to be running NixOS. Attempting to patch"
+        print(" ".join([nix_os_msg, fname]))
```
Comment (kennytm) [inline src/bootstrap/bootstrap.py]: Why don't we simply write `print(nix_os_msg, fname)` here?

**Comment by zackmdavis** [comment]
(not a core reviewer, but commenting anyway because Python!!)

> do you like to include some unit tests with pytest?

If there were to be unit tests, I'd imagine they should be with standard-libary `unittest` rather than `pytest`. (Python's batteries-included philosophy is part of what makes it such a good scripting-glue language; it's better if Rust's build process doesn't depend on getting a third-party library from PyPI.)

> should be converted to a function because it doesn't use `self` anywhere

Or `@staticmethod`?

**Comment by Mark-Simulacrum** [comment]
These changes primarily look good to me. I'll r? @alexcrichton for approval though.

> Where can I find the unit tests for this script? In case it doesn't exist yet, do you like to include some unit tests with pytest?

I don't know if there's much in bootstrap.py that really could be unit tested -- most of it is setup code interacting with the network or file system in preparation for handing off to rustbuild itself. I'm not opposed to adding tests for the parts we can test, though!

> Some methods like fix_executable, get_string, and exe_suffix in the RustBuild class should be converted to a function because it doesn't use self anywhere. What do you think?

Seems like a good idea, though not familiar with python overly much myself. Feel free to add that here if you'd like!

**Review by Mark-Simulacrum — APPROVED** [review]

**Comment by alexcrichton** [comment]
@bors: r+

We always welcome refactorings of our Python, as you can probably tell we in general aren't the most seasoned Python programmers, so help is always very much appreciated!

**Pull request merged**